src/Goat/Update.elm

```diff
@@ -369,6 +369,25 @@ finishValidDrawing model drawingInfo =
             model
 
 
+foldDistance : Position -> ( Float, Position ) -> ( Float, Position )
+foldDistance position ( distance, previousPosition ) =
+    distance
+        + (calcDistance position previousPosition)
+        => position
+
+
+finishFreeHandDrawing : EditState -> DrawingInfo -> Model -> Model
+finishFreeHandDrawing finishedEditState ({ start, curPos, positions } as drawingInfo) model =
+    let
+        ( distance, _ ) =
+            List.foldl foldDistance ( 0.0, start ) (positions ++ [ curPos ])
```

Comment:
would be interesting to [benchmark](http://package.elm-lang.org/packages/BrianHicks/elm-benchmark/latest) whether it's faster to do what you did or:
```elm
List.foldr foldDistance ( 0.0, start ) (curPos :: (List.reverse positions))
```